text/0000-standardize-error-docs.md

```diff
@@ -0,0 +1,55 @@
+
+- Firstly, that the style of error docs is not defined by RFC but by a "style spec" located in the `rust-lang/rust` repository. This style spec can be updated from time-to-time in order to ensure consistent formatting across error docs. Any change to the style spec must be with the consensus of the dev-tools team. It is recommended that the style spec is initially extracted from [RFC 1567].
+- Secondly, that the idea of "labels" is introduced to error docs. These would specify certain attributes about an error code that are important for users to know. Error codes can be marked with labels easily and are verified at compile-time (this would probably happen in the `register_diagnostics` macro). These labels would be added as markdown headers (and any other potential places) at doc-time and would replace the current typo-prone system of copying long lines around. Labels are defined in the style spec. Three  label-candidates are exampled here:
+  - `removed`: An error code has been removed, and therefore the labeled docs docs exist for backwards-compatibility purposes only. The docs may also contain a note explaining *where* the error code was moved to, if at all. (The style spec can specify this more fully)
+  - `internal`: An error code is internal to the compiler/standard library. This means that it can only occur when using a perma-unstable feature gate which is explicitly marked as only intended for use internally.
+  - `feature_gated`: An error code can only be emitted when using an unstable feature. This error is subject to change or removal, all the normal feature gate non-guarantees apply.
+- Thirdly, that error docs (including the new labels feature) are aggressively linted according to the style spec. This helps keep the error docs consistently styled.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+It requires a lot of effort:
+- The error docs check in `tidy` will have to be kept up to date with the style spec. 
+- Error docs will have to be kept up to date with the style spec.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Some changes must be made, otherwise PR authors coming up with their own formatting for error docs will continue to increase. *This is already happening* and only makes error docs harder to understand.
+
+# Prior art
+[prior-art]: #prior-art
+
+[RFC 1567] is prior art, its significance to this RFC was mentioned in the [motivation] section.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Is the current `mdBook` format used to present error docs still sufficient? Its index page is very unappealing and makes it hard to quickly jump to an error. Furthermore, it does not interface well with the new idea of error doc labels and cannot quickly show labels applied to an error.
+- Should changes to the style spec require more or less than "dev-tools team consensus"?
```

Comment:
It's actually two questions:

 1. Should changes to the style spec require more or less a "team" consensus?
 2. Which team should be in charge of error codes?

Reply:
1. Hmm, perhaps a good standard would be "changes that would in turn require modifications to the docs" (so then spelling changes, etc to the spec would be fine)?

2. As for the team, I obviously agree that dev-tools is wrong. I also am somewhat opposed to wg-diagnostics having anything to do with it, they seem focused on changes to error messages themselves, no docs. Worst case, the docs team is reformed or something.

Reply:
> the docs team is reformed or something.

It's complicated. ^^'